resources/cloudfront_distributions.go

```diff
@@ -0,0 +1,594 @@
+			},
+			// Restrictions start
+			{
+				Name:     "restrictions_geo_restriction_restriction_type",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Restrictions.GeoRestriction.RestrictionType"),
+			},
+			{
+				Name:     "restrictions_geo_restriction_restriction_items",
+				Type:     schema.TypeStringArray,
+				Resolver: schema.PathResolver("Restrictions.GeoRestriction.Items"),
+			},
+			// Restrictions End
+			{
+				Name: "status",
+				Type: schema.TypeString,
+			},
+			//ViewerCertificate start
+			{
+				Name:     "viewer_certificate_acm_certificate_arn",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.ACMCertificateArn"),
+			},
+			{
+				Name:     "viewer_certificate_certificate",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.Certificate"),
+			},
+			{
+				Name:     "viewer_certificate_certificate_source",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.CertificateSource"),
+			},
+			{
+				Name:     "viewer_certificate_cloudfront_default_certificate",
+				Type:     schema.TypeBool,
+				Resolver: schema.PathResolver("ViewerCertificate.CloudFrontDefaultCertificate"),
+			},
+			{
+				Name:     "viewer_certificate_iam_certificate_id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.IAMCertificateId"),
+			},
+			{
+				Name:     "viewer_certificate_minimum_protocol_version",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.MinimumProtocolVersion"),
+			},
+			{
+				Name:     "viewer_certificate_ssl_support_method",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ViewerCertificate.SSLSupportMethod"),
+			},
+			//ViewerCertificate end
+			{
+				Name:     "web_acl_id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("WebACLId"),
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:     "aws_cloudfront_distribution_cache_behaviours",
+				Resolver: fetchCloudfrontDistributionCacheBehaviours,
+				Columns: []schema.Column{
+					{
+						Name:     "distribution_id",
+						Type:     schema.TypeUUID,
+						Resolver: schema.ParentIdResolver,
+					},
+					{
+						Name: "path_pattern",
+						Type: schema.TypeString,
+					},
+					{
+						Name: "target_origin_id",
+						Type: schema.TypeString,
+					},
+					{
+						Name: "viewer_protocol_policy",
+						Type: schema.TypeString,
+					},
+					{
+						Name:     "allowed_methods",
+						Type:     schema.TypeStringArray,
+						Resolver: schema.PathResolver("AllowedMethods.Items"),
+					},
+					{
+						Name:     "cached_methods",
+						Type:     schema.TypeStringArray,
+						Resolver: schema.PathResolver("AllowedMethods.CachedMethods.Items"),
+					},
+				},
+			},
+			{
+				Name:     "default_cache_behaviour_lambda_function_associations", //todo check if this kind of relation is correct
```

Comment:
remove todo, and  I would do cache_behaviour  instead of default_cache_behaviour as changed above with the fields